command/agent.go

```diff
@@ -490,6 +496,152 @@ func (c *AgentCommand) Run(args []string) int {
 			return 1
 		}
 
+		// Configure persistent storage and add to LeaseCache
+		if config.Cache.Persist != nil {
+			if config.Cache.Persist.Path == "" {
+				c.UI.Error("must specify persistent cache path")
+				return 1
+			}
+
+			// Set AAD based on key protection type
+			var aad string
+			switch config.Cache.Persist.Type {
+			case "kubernetes":
+				aad, err = cacheboltdb.GetServiceAccountJWT(config.Cache.Persist.ServiceAccountTokenFile)
```

Comment:
The `cacheboltdb` package does not need to be aware of where/how it's getting this aad value so maybe turn this into a private func, `getServiceAccountJWT`, that lives directly within this package.

Reply:
Done in cee5bfb